### PR TITLE
Handle variable length records

### PR DESCRIPTION
**Pull request by kbatyr**

### CODE REVIEW

File: decoders/netflow/packet.go [hunk 1]
```diff
@@ -114,6 +114,22 @@ type Field struct {
 
 	// The length (in bytes) of the field.
 	Length uint16
+
+	// IANA private enterprise number (PEN) of the authority defining the
+	// Information Element identifier in this Template Record.
```
Comment (yerden) [inline decoders/netflow/packet.go:119]: I'd add that it's only valid for IPFIX. Well, it's up to you.

File: decoders/netflow/netflow.go [hunk 2]
```diff
@@ -184,6 +195,31 @@ func DecodeDataRecordFields(payload *bytes.Buffer, listFields []Field, record *[
...
 }
 
+func getVariableLength(payload *bytes.Buffer) (int, bool) {
+	b, err := payload.ReadByte()
+	if err != nil {
+		return 0, false
+	}
+
+	// RFC 7011 Sec. 7.Variable-Length Information Element.
+	if b < 0xff {
```
Comment (yerden) [inline decoders/netflow/netflow.go]: `b != 0xff` is more efficient though.

File: decoders/netflow/packet.go [hunk 3]
```diff
@@ -114,6 +114,22 @@ type Field struct {
...
+
+// IsVariableLength reports whether the Field is variable-length Information
+// Element (RFC 7011 sec. 3.2. Field Specifier Format).
+func (f *Field) IsVariableLength() bool {
+	return f.Length == 0xffff
+}
+
+// IsEnterpriseNumber reports whether the Field is an enterprise-specific
+// Information Element (RFC 7011 sec. 3.2. Field Specifier Format).
+func (f *Field) IsEnterpriseNumber() bool {
```
Comment (yerden) [inline decoders/netflow/packet.go]: `IsEnterpriseNumber` is misleading. I'd say it's `IsEnterpriseSpecific`.

Comment (yerden) [reply decoders/netflow/packet.go]: Or, `ContainsEnterpriseNumber` if you will.

File: decoders/netflow/packet.go (hunk 1 above)
Comment (kbatyr) [reply decoders/netflow/packet.go:119]: It's better to write clearly, thanks, added documention.

File: decoders/netflow/netflow.go (hunk 2 above)
Comment (kbatyr) [reply decoders/netflow/netflow.go]: fixed

File: decoders/netflow/packet.go (hunk 3 above)
Comment (kbatyr) [reply decoders/netflow/packet.go]: I think `ContainsEnterpriseNumber` sounds better, renamed

**Review by yerden — APPROVED** [review]

**Review by yerden — APPROVED** [review]

**Pull request merged**